extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/DbCheckingPMetaSupplier.java

```diff
@@ -0,0 +1,195 @@
+    private final DataConnectionRef dataConnectionRef;
+    private int preferredLocalParallelism = Vertex.LOCAL_PARALLELISM_USE_DEFAULT;
+
+    private transient Address ownerAddress;
+
+    /**
+     * Creates a new instance of this meta supplier.
+     */
+    public DbCheckingPMetaSupplier(@Nullable Permission requiredPermission,
+                                   boolean shouldCheck,
+                                   boolean forceTotalParallelismOne,
+                                   @Nullable String databaseName,
+                                   @Nullable String collectionName,
+                                   @Nullable SupplierEx<? extends MongoClient> clientSupplier,
+                                   @Nullable DataConnectionRef dataConnectionRef,
+                                   @Nonnull ProcessorSupplier processorSupplier) {
+        this.requiredPermission = requiredPermission;
+        this.shouldCheck = shouldCheck;
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        this.databaseName = databaseName;
+        this.collectionName = collectionName;
+        this.processorSupplier = processorSupplier;
+        this.clientSupplier = clientSupplier;
+        this.dataConnectionRef = dataConnectionRef;
+    }
+
+    /**
+     * Sets preferred local parallelism. If {@link #forceTotalParallelismOne} is selected, this
+     * method will have no effect.
+     */
+    public DbCheckingPMetaSupplier withPreferredLocalParallelism(int preferredLocalParallelism) {
+        this.preferredLocalParallelism = forceTotalParallelismOne ? 1 : preferredLocalParallelism;
+        return this;
+    }
+
+    @Override
+    public int preferredLocalParallelism() {
+        return forceTotalParallelismOne ? 1 : preferredLocalParallelism;
+    }
+
+    @Nullable
+    @Override
+    public Permission getRequiredPermission() {
+        return requiredPermission;
+    }
+
+    /**
+     * If true, only one instance of given supplier will be created.
+     */
+    public DbCheckingPMetaSupplier forceTotalParallelismOne(boolean forceTotalParallelismOne) {
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        return this;
+    }
+
+    @Override
+    public boolean initIsCooperative() {
+        return true;
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        if (forceTotalParallelismOne) {
+            preferredLocalParallelism = 1;
+            if (context.localParallelism() != 1) {
+                throw new IllegalArgumentException(
+                        "Local parallelism of " + context.localParallelism() + " was requested for a vertex that "
+                                + "supports only total parallelism of 1. Local parallelism must be 1.");
+            }
+            String key = getPartitionKey(newUnsecureUuidString());
+            int partitionId = context.hazelcastInstance().getPartitionService().getPartition(key).getPartitionId();
```

Comment:
do you just select a random member here in this convoluted way, or this something more?

Reply:
Yes, code is the same as in ProcessorMetaSupplier#forceTotalParallelismOne

Reply:
and which member do you want? any? or truly random, different for each execution? if any=don't care, then use coordinator via `forceTotalParallelismOne` with local member address (similar to `CreateTopLevelDagVisitor` does).

Reply:
now the code reuses forceTotalParallelismOne via composition